src/components/seeds/database/index.tsx

```diff
@@ -375,18 +378,33 @@ export default function Database(props: DatabaseProps): JSX.Element {
           setFieldOptions(allValues);
         }
       };
-      if (searchCriteria) {
-        populateSearchResults();
+
+      // If we are loading search results because the locale changed, we need to reset the search
+      // criteria since they might contain localized values from the old locale. In that case, we
+      // need to skip populating the search results based on the old criteria, but we'll be called
+      // again as an effect of the setCriteria() call.
+      if (!searchedLocaleRef.current) {
+        // First rendering pass was before strings were loaded, so no need to reset criteria.
+        searchedLocaleRef.current = loadedStringsForLocale;
       }
-      populateAvailableFieldOptions();
 
-      populateFieldOptions();
+      if (searchedLocaleRef.current === loadedStringsForLocale) {
+        if (searchCriteria) {
+          populateSearchResults();
+        }
+        populateAvailableFieldOptions();
+
+        populateFieldOptions();
+      } else {
+        searchedLocaleRef.current = null;
+        setSearchCriteria({ ...DEFAULT_SEED_SEARCH_FILTERS });
+      }
```

Comment:
This feels a little inelegant to me, but I couldn't come up with a nicer-looking way to do it that didn't end up causing duplicate API requests on locale change.

Reply:
This lookg good to me, but maybe @karthikbtf or @charlie-tf have other suggestion